### PR TITLE
feat: Automated releases w/ conventional commits; Prep for npm package publish of calc and api-sidecar;

### LINKED ISSUE #103: Automate versioning
Investigate and implement automated versioning. A service [like this](https://github.com/release-it/release-it) may be useful.

### LINKED ISSUE #284: Investigate creating NPM package for `calc` crate
**Proposed Change or Idea**

Sidecar has a Rust crate that uses was-pack to expose JS bindings. It should be investigated if the crate should be published to crates.io and if the package should be published to NPM.

### LINKED ISSUE #285: Look into publishing Sidecar as NPM package
**Proposed Change or Idea**

Look into publishing Sidecar as an NPM package and creating a release workflow that helps facilitate it.

### PR DESCRIPTION
**Pull request by emostov**
Closes #284, closes #285, closes #103 

Changes:
- introduction of `deploy` script that updates the CHANGELOG based on commit history. (All commits now need to use conventional commit format in order to count in the CHANGELOG). Bumps the version based on commits, and helps publish to NPM.
- Updates to `package.json` to enable our package to work better with the npm registry.
- Published the @substrate/calc package and now include the package from the npm registry instead of building it everytime sidecar starts. We need to do this because the pre/post install hooks are not compatible with the download process of an npm package. Further, it would require people using the package to have rust installed with wasm-pack, which we don't want.
- Updates to the README to reflect 1) no longer need to install Rust for normal usage 2) how to start the npm package
- Other README updates:
    - Remove paths and instead link to full docs
    - HTML and badges so the README looks pretty 
    - How to cut a new release
- For the NPM package I was having issues reading in the `specs.yml` file. The solution was to put in the absolute path, however, the github workflow for running tests broke because it could not find the `specs.yml` file with the absolute path. Instead of sinking further time into that I issue I decided to use a factory method to create the specs in TS. I think this has the benefit of essentially being guaranteed to work across environments since its just a normal class and not some external file that needs to be read in.

Misc:
- the license for the calc crate is Apache 2, not sure if we want to change that (note: remember to change license badge in README if this is changed)
- we need to bump the version of the calc crate a minor since npm won't let us publish the same version twice

To use this package users would do a global install of @substrate/api-sidecar, and then could start the server by running `substrate-api-sidecar` at the command line.

Follow up:
- [ ] NPM publish @substrate/calc
- [ ] NPM publish @substrate/api-sidecar
- [ ] Improve README for calc package with better documentation (#297)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
+    </a>
+  </p>
+</div>
+
+<br /><br />
+
+## Quick start for @substrate/api-sidecar CLI npm package
+
+```bash
+# Globally install the package
```
Comment (dvdplm) [inline README.md:26]: Does it have to be installed in the global registry? If yes, let's mention that in the comment. If not, mention that too. :)

File: README.md [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
-[up-to-date version of `rustup`](https://www.rust-lang.org/tools/install) installed to manage Rust
+### Quick install
+
+Simply run `yarn`.
+
+### Rust development installation
+
+If you are looking to hack on the `calc` Rust crate:
+
+Make sure your machine has an [up-to-date version of `rustup`](https://www.rust-lang.org/tools/install) installed to manage Rust
```
Comment (dvdplm) [inline README.md:47]:
```suggestion
If you are looking to hack on the `calc` Rust crate make sure your machine has an [up-to-date version of `rustup`](https://www.rust-lang.org/tools/install) installed to manage Rust
```

File: README.md [hunk 3]
```diff
@@ -110,81 +148,7 @@ CALC_DEBUG=1 yarn
 
 ## Available paths
```
Comment (dvdplm) [inline README.md:149]:
```suggestion
## Available endpoints
```

File: README.md [hunk 4]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
 ## Contribute
 
 Need help or want to contribute ideas or code? Head over to our [CONTRIBUTING](CONTRIBUTING.md) doc for more information.
+
+## Note for maintainers
+
+All the commits in this repo follow the Conventional Commits spec. When merging a PR, make sure 1/ to
+use squash merge and 2/ that the title of the PR follows the Conventional Commits spec. This will
```
Comment (dvdplm) [inline README.md:196]: Link to Conventional Commits spec please.

File: README.md [hunk 5]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
...
+make squash the PR into one conventional commit.
```
Comment (dvdplm) [inline README.md:197]: This needs rephrasing.

File: README.md [hunk 6]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
...
+
+The history of commits will be used to generate the `CHANGELOG`. To do so, run yarn deploy on the master
```
Comment (dvdplm) [inline README.md:199]: I really like https://keepachangelog.com/en/1.0.0/ and try to promote using it everywhere. Not sure how close that is to what is generated though.

File: README.md [hunk 7]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
...
+branch. This command will look at all the commits since the latest tag, bump the package version according
+to semver rules, and generate a new `CHANGELOG`.
+
+There might be special cases where you don't want to follow semver. In this case, run `yarn deploy -r {major,minor,patch}`.
```
Comment (dvdplm) [inline README.md:203]: 😨

File: README.md [hunk 8]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
...
+
+The above command, which only does local operations and doesn't push anything, will output more or
+less the following lines:
+
+``` bash
+$ yarn deploy
```
Comment (dvdplm) [inline README.md:209]: "The above command" seems to refer to `yarn deploy -r {major,minor,patch}` but I think you mean `yarn deploy` here.

File: README.md [hunk 9]
```diff
@@ -225,3 +189,33 @@ curl -s http://0.0.0.0:8080/block | jq
...
+$ rimraf lib/ && tsc
+✔ bumping version in package.json from 0.18.1 to 0.18.2
+✔ outputting changes to CHANGELOG.md
+✔ committing package.json and CHANGELOG.md
+✔ tagging release v0.18.2
+ℹ Run `git push --follow-tags origin master && npm publish` to publish
+```
+
+To publish the new package, just follow the instructions: `git push --follow-tags origin master && npm publish.` 
+You must have access to the @substrate organization on npm to be able to publish.
```
Comment (dvdplm) [inline README.md:221]:
```suggestion
You must have access to the @substrate organization on npm to publish.
```

File: calc/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      <img alt="apache-2.0" src="https://img.shields.io/badge/License-Apache%202.0-blue.svg" />
+    </a>
+  </p>
+</div>
+
+<br /><br />
+
+## About
+
+This package is generated from a Rust crate using `wasm-bindgen` and was initially developed
```
Comment (dvdplm) [inline calc/README.md:24]: link to the rust crate

File: calc/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+solely to use as a dependency for `substrate-api-sidecar`. We are now offering this package as a
+standalone through the npm registry.
+
+## Usage
+
+Example usage for the package can be found in Sidecar's [staking payout service](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/services/accounts/AccountsStakingPayoutsService.ts) and Sidecar's [block service](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/services/blocks/BlocksService.ts)
+
+## Contributing
+
+We welcome [contributions for documentation and code](https://github.com/paritytech/substrate-api-sidecar/pulls). If you have any questions you can reach the maintainers by [filing an issue on github.](https://github.com/paritytech/substrate-api-sidecar/issues)
```
Comment (dvdplm) [inline calc/README.md:34]: Shouldn't contributions to `calc` go to the calc crate directly?

**Review by dvdplm — APPROVED** [review]
lgtm modulo nitpicks

File: README.md (hunk 6 above)
Comment (emostov) [reply README.md]: What is generated looks like this: https://github.com/paritytech/txwrapper/blob/master/CHANGELOG.md

I think it is very close to that except there is no unreleased section.

File: README.md [hunk 12]
```diff
@@ -1,13 +1,99 @@
...
+This service requires Node version 12 or higher.
+
+## Table of contents
+
+- [NPM package installation and usage](#npm-package-installation-and-usage)
+- [Source code installation and usage](#source-code-installation-and-usage)
+- [Running](#running)
+- [Configuration](#configuration)
+- [Debugging fee and payout calculations](#debugging-fee-and-payout-calculations)
+- [Available endpoints](https://paritytech.github.io/substrate-api-sidecar/dist/)
```
Comment (emostov) [inline README.md:42]: We could link to within the README for continuity, but that section just has the same link

**Pull request merged**